index.js

```diff
@@ -0,0 +1,92 @@
+     * @method jenkinsCommand
+     * @param  {Object}   options            An object that tells what command & params to run
+     * @param  {String}   options.module     Jenkins client module. For example: job, build
+     * @param  {String}   options.action     Jenkins client action in the given module. For example: get, create
+     * @param  {Object}   options.params     Parameters to run with
+     * @param  {Function} callback           fn(err) from jenkinsClient
+     */
+    _jenkinsCommand(options, callback) {
+        // To pass arguments as an array, we need to use apply
+        options.params.push(callback);
+
+        return this.jenkinsClient[options.module][options.action].apply(
+            this.jenkinsClient[options.module],
+            options.params);
+    }
+
+    /**
+     * Constructor
+     * @method constructor
+     * @param  {Object} options           Configuration options
+     * @param  {Object} options.host      Jenkins hostname to make requests to
+     * @param  {Object} options.username  Jenkins username
+     * @param  {Object} options.password  Jenkins password/token
+     */
+    constructor(options) {
+        super();
+        this.host = options.host;
+        this.username = options.username;
+        this.password = options.password;
+        // need to pass port nubmer in the future
+        this.baseUrl = `http://${this.username}:${this.password}@${this.host}:8080`;
+        this.jenkinsClient = jenkins({
+            baseUrl: this.baseUrl,
+            crumbIssuer: true
+        });
+
+        /* eslint-disable no-underscore-dangle */
+        this.breaker = new Breaker(this._jenkinsCommand.bind(this));
+    }
+
+    /**
+     * Create a jenkins job and start the build
+     * @method start
+     * @param  {Object}   config            A configuration object
+     * @param  {String}   config.buildId    ID for the build and also name of the job in jenkins
+     * @param  {String}   config.jobId      ID for the job
+     * @param  {String}   config.pipelineId ID for the pipeline
+     * @param  {String}   config.container  Container for the build to run in
+     * @param  {String}   config.scmUrl     Scm URL to use in the build
+     * @param  {Function} callback          fn(err) where data is always null
+     */
+    _start(config, callback) {
+        let xml;
+        let jobName;
+
+        async.waterfall([
+            (next) => {
+                const configPath = path.resolve(__dirname, './config/test-job.xml');
+
+                fs.readFile(configPath, 'utf-8', (err, fileContents) => {
+                    xml = fileContents;
+                    jobName = config.buildId;
+                    next(err);
+                });
+            },
+            (next) => this.breaker.runCommand({
+                module: 'job',
+                action: 'create',
+                params: [jobName, xml]
+            }, next),
+            (data, next) => this.breaker.runCommand({
```

Comment:
since it doesn't seem like you use the data object, does it make more sense to use `async.series` instead of `async.waterfall`  ?